### ISSUE TITLE
Support forward secrecy

### ISSUE THREAD

**Issue by BamButz** [caprover/caprover#371]
**Describe the solution you'd like**
Support forward secrecy for improved data security so that when the private key gets compromised it cannot be used to decrypt recorded SSL traffic.

E.g. by extending the standard application NGINX configuration to only use cipher suites that support forward secrecy.

Recommended cipher suites can be found on https://wiki.mozilla.org/Security/Server_Side_TLS

**Describe alternatives you've considered**
Modify every app's NGINX configuration by hand.

E.g. extend
```
listen              443 ssl;
ssl_certificate     <%-s.crtPath%>;
ssl_certificate_key <%-s.keyPath%>;
```
with
```
ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
```

**Additional context**

Currently capped to grade B because of lack of forward secrecy.
![SSL Test](https://user-images.githubusercontent.com/7022144/52637367-59ba4b80-2ecf-11e9-8c89-6967aa2dc624.png)
From: https://www.ssllabs.com/ssltest

**Comment by githubsaturn** [comment]
Good suggestion, will change to Mozilla Intermediate Profile:

```
    # intermediate configuration. tweak to your needs.
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS';
    ssl_prefer_server_ciphers on;
```

**Comment by githubsaturn** [comment]
Will be released in v1.3
https://github.com/caprover/caprover/commit/63316974e789bd8b1d7bc539a57d9935587d8b75
